htm/src/nupic/types/Sdr.hpp

```diff
@@ -27,7 +27,7 @@
 #include <vector>
 #include <numeric>
 #include <nupic/types/Types.hpp>
-#include <nupic/ntypes/Array.hpp>
+// #include <nupic/ntypes/Array.hpp>   // TODO: causes a dependancy on netapi
```

Comment:
> We aim to make SDR part of "core" only, not depending on ntypes etc..

Well, the NetworkAPI is just as much part of the 'core' as the htm algorithms.  But this is the only part of the htm that references anything in the NetworkAPI.  

Since the SDR class performs a copy anyway (actually a conversion), we could change the interface between Array and SDR to be passing a raw pointer to the buffer.  Perhaps a better alternative is to have the Array 'contain' the SDR.  A sending algorithm could pass its SDR intact to a receiving algorithm without incurring a copy.  I would have to make some changes to the Array but that makes the most since to me.